### ISSUE TITLE
[Performance]: Slowdown compared to Gradio

### ISSUE THREAD

**Issue by theoren** [vllm-project/vllm#8866]
### Proposal to improve performance

vLLM is amazingly fast
However, when running below prompt, with meta-llama/Meta-Llama-3-8B-Instruct, Gradio takes ~4sec per prompt (one by one) while vLLM takes ~12sec by def. When setting --quantization fp8 times reduced to ~8s
Overall vLLM is much faster since it allows to process in parallel while Gradio doesn't
Tested with AWS L4, Gradio 4.43.0
What am I missing?

`prompt = """You are a knowledgeable, efficient, and direct Al assistant. Provide concise answers up to 100 words`
`without explainations or extra notes, focusing on the key information needed. Answer in question: answer JSON format`
`**User:**I like the color red. Our website is www.nba.com. My age is 18.`
`**Assistant:**Great. Write 3 things for me to answer.`
`**User:**What is our website? What is my age? What kind of drink do I like to drink?`
`**Assistant**:`
`"""`

### Your current environment (if you think it is necessary)

```text
The output of `python collect_env.py`
```

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by mgoin** [comment]
How does Gradio host models? Could you possible share the scripts/commands you have for running vLLM vs Gradio?
